elasticsearch.md

```diff
@@ -799,11 +799,86 @@ So that Kibana will forward the necessary headers to Elasticsearch.
 
 [Impersonation](details/impersonation.md) is supported by this rule without an extra configuration.
 
-#### `groups_or` (or `groups`)
+#### `user_belongs_to_groups`
 
-`groups_or: ["group1", "group2"]`
+The ACL block will match when the user belongs to groups matching the condition defined in the `user_belongs_to_groups` config section. The information about what users belong to what groups is defined in the `users` section, typically situated after the ACL, further down in the YAML.
 
-The ACL block will match when the user belongs to any of the specified groups. The information about what users belong to what groups is defined in the `users` section, typically situated after the ACL, further down in the YAML.
+##### `any_of`
+
+The ACL block will match when the user belongs to any of the specified groups.
+
+```yaml
+  user_belongs_to_groups:
+    any_of: ["group1", "group2"]
+```
+
+With deprecated syntax (without being wrapped inside the `user_belongs_to_groups` section):
+- `groups_or: ["group1", "group2"]`
+- `groups: ["group1", "group2"]`
+
+##### `all_of`
+
+This rule is very similar to the above defined `any_of` rule, but this time ALL the groups listed in the array are required (boolean AND logic), as opposed to at least one (boolean OR logic) of the `any_of` rule.
+
+```yaml
+  user_belongs_to_groups:
+    all_of: ["group1", "group2"]
+```
+
+With deprecated syntax (without being wrapped inside the `user_belongs_to_groups` section):
+- `groups_and: ["group1", "group2"]`
+
+##### `not_any_of`
+
+The ACL block will match when the user belongs to NONE of the specified groups.
+
+Looking at the example below:
+- ACL block will match for user that belongs to `group0`
+- ACL block will NOT MATCH for user that belongs only to `group1`
+- ACL block will NOT MATCH for user that belongs only to `group2`
+- ACL block will not match for user that belongs to both `group1` and `group2`
```

Comment:
should we add the example with user with groups `group0` & `group1` & `group2`?

Reply:
Added for both any_of and all_of